### ISSUE TITLE
[Bug] Periodic CPU spike when there are too many connections

### ISSUE THREAD

**Issue by HarukaMa** [AleoHQ/snarkOS#1632]
## 🐛 Bug Report

I'm trying to bump the max nodes count to 100 on my operator node and found that there are strange CPU spikes. I've failed to observe a fixed interval for it.

I've also changed the thread count in `main.rs` to make it always uses all available cores (basically using sync node settings), except the blocking threads are the same as core count to make profiling data easier to read.

I'm not using the official codebase but I think it's still relevant.

## Steps to Reproduce

1. Connect to like 100 peers
2. Observe that sometimes snarkOS uses 100% of CPU

## Expected Behavior

Ideally peering shouldn't be that resource intensive.

## Your Environment

- snarkOS 2.0.1, but on my branch (HarukaMa/custom-pool-staging)
- rust 1.58
- Debian sid

I'm running a profiler to see what's actually using all CPUs.

**Comment by HarukaMa** [comment]
![image](https://user-images.githubusercontent.com/861659/153777585-3c898514-a530-4850-bc3e-d1193cd7966c.png)

It looks like almost all CPU time is used to verify PoSW proofs. From the known function calls, they most probably come from deserializing block headers.

I have tried to print some logs around `snarkvm_dpc::block::block::Block<N>::is_valid` and `snarkvm_dpc::block::header::BlockHeader<N>::is_valid`, and I've found that it's still related to `snarkos::network::ledger::Ledger<N, E>::update_peer`: There are 64 linear block headers inside each block locator which means there will be 64 block headers to deserialize. During each deserialization, the PoSW proof inside the block header will be verified. Say each header verification takes about 35 ms (which is what I'm reading from my node), it would take about 2.24 seconds to verify all of the headers. For 100 peers, that will be 224 seconds. Say the machine has 32 threads, that'll be 7 seconds in total. (In my case my test machine only has 12 threads so it takes almost 20 seconds to verify all of them), not counting other things to do like processing new blocks and verifying incoming proofs from provers.

I'm not sure if this is intended or connecting to more peers is actively discouraged, but IMO this process should not take that much CPU resources.
